source/agora/crypto/Hash.d

```diff
@@ -120,60 +121,36 @@ public Hash hashFull (T) (scope const auto ref T record)
 /// Ditto
 public void hashPart (T) (scope const auto ref T record, scope HashDg state)
     /*pure*/ nothrow @nogc
-    if (is(T == struct))
 {
-    static if (is(typeof(T.init.computeHash(HashDg.init))))
-        record.computeHash(state);
-    else static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
-        state(record[]);
+    static if (is(T == struct))
+    {
+        static if (is(typeof(T.init.computeHash(HashDg.init))))
+            record.computeHash(state);
+        else static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
+            doHashWithSerializer(record[], state);
+        else
+            foreach (const ref field; record.tupleof)
+                doHashWithSerializer(field, state);
```

Comment:
Do we even need the `foreach` here ? Can't we just fwd to the serializer ?

Reply:
Updated.

Reply:
Still need `foreach` as there may be a struct within a record which needs to be hashed with computeHash. However it should call the hashPart again to handle this correctly. Now updated again.